### PR TITLE
operator: fix the process to check if an operator is ready

### PR DESCRIPTION
**Pull request by jmontesi**
The two main modifications are:

* Only wait for an operator to be ready when it's in a state that could potentially lead to a Succeeded state. For instance, it does not make sense to wait when it's in the Failed state.

* Do not log with logrus.Faltalf() the timeout condition, as this causes the CNF Certification Suite to return with an error exit status. It should set the operator as non compliant but continue with the remaining tests.

The combination of these two points cause the CNF Certification Suite to fail and stop executing if an operator never reaches the Succeeded state, when only a test case failure should occur.

The only test case affected is operator-install-status-succeeded.

### CODE REVIEW

File: cnf-certification-test/operator/phasecheck/phasecheck.go [hunk 1]
```diff
@@ -18,44 +18,66 @@ package phasecheck
...
+			logrus.Errorf(errMsg)
+			tnf.ClaimFilePrintf(errMsg)
+			return false
 		}
-		isReady = IsOperatorPhaseSucceeded(freshCsv)
-		logrus.Debugf("Waiting for %s to be in Succeeded phase: %s", provider.CsvToString(freshCsv), freshCsv.Status.Phase)
+
+		// update old csv and check status again
+		*csv = *freshCsv
+		if csv.Status.Phase == v1alpha1.CSVPhaseSucceeded {
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply cnf-certification-test/operator/phasecheck/phasecheck.go:50]: That could lead to an uncontrolled recursion whereby `IsOperatorPhaseSucceeded()` and `waitOperatorReady()` called each other constantly. But I think the failed states should be considered at that point to leave the function as early as possible without waiting for the timeout.

[the remark this answers is not in the document]
Comment (edcdavid) [reply cnf-certification-test/operator/phasecheck/phasecheck.go]: why not just call waitOperatorReady here and use IsOperatorPhaseSucceeded to just check if operator has succeeded (with no recursive wait call)?

**Comment by bnshr** [comment]
When the operator is in failed state, we do not retry during the timeout duration in this solution. Is that right?

**Comment by jmontesi** [comment]
> When the operator is in failed state, we do not retry during the timeout duration in this solution. Is that right?

Right.

File: cnf-certification-test/operator/phasecheck/phasecheck.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmontesi) [reply cnf-certification-test/operator/phasecheck/phasecheck.go]: I just refactored the code. Let me know if you prefer this new implementation.

**Review by edcdavid — APPROVED** [review]

**Review by greyerof — APPROVED** [review]

**Pull request merged**